rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/ContextMenu.java

```diff
@@ -199,4 +194,159 @@ private void enableItems(MenuItem... items) {
+
+    }
+
+    private static class NodeSourceMenu {
+        private RMController controller;
+
+        private ContextMenu menu;
+
+        private String nodeSourceName;
+
+        private MenuItem deployItem;
+
+        private MenuItem undeployItem;
+
+        private MenuItem editItem;
+
+        private NodeSourceMenu(RMController controller, ContextMenu menu, String nodeSourceName) {
+            this.controller = controller;
+            this.menu = menu;
+            this.nodeSourceName = nodeSourceName;
+        }
+
+        private MenuItem getDeployItem() {
+            return this.deployItem;
+        }
+
+        private MenuItem getUndeployItem() {
+            return this.undeployItem;
+        }
+
+        private MenuItem getEditItem() {
+            return this.editItem;
+        }
+
+        private NodeSourceMenu build() {
+            this.deployItem = new MenuItem("Deploy", this.menu.deployItemImageResource);
+            this.deployItem.addClickHandler(onClick -> this.controller.deployNodeSource());
+
+            this.undeployItem = new MenuItem("Undeploy", this.menu.undeployItemImageResource);
+            this.undeployItem.addClickHandler(onClick -> this.controller.undeployNodeSource());
+
+            this.editItem = new MenuItem("Edit", this.menu.editItemImageResource);
+            NodeSourceStatus nodeSourceStatus = this.menu.nodesource == null ? null
+                                                                             : this.menu.nodesource.getNodeSourceStatus();
+            this.editItem.addClickHandler(onClick -> this.controller.editNodeSource(this.nodeSourceName,
+                                                                                    nodeSourceStatus));
+            return this;
+        }
+
+    }
+
+    private static class ExportMenu {
+
+        private RMController controller;
+
+        private ContextMenu menu;
+
+        private String nodeSourceName;
+
+        private MenuItem exportNodeSourceItem;
+
+        private MenuItem exportInfrastructureItem;
+
+        private MenuItem exportPolicyItem;
+
+        private ExportMenu(RMController controller, ContextMenu menu, String nodeSourceName) {
+            this.controller = controller;
+            this.menu = menu;
+            this.nodeSourceName = nodeSourceName;
+        }
+
+        private MenuItem getExportNodeSourceItem() {
+            return this.exportNodeSourceItem;
+        }
+
+        private MenuItem getExportInfrastructureItem() {
+            return this.exportInfrastructureItem;
+        }
+
+        private MenuItem getExportPolicyItem() {
+            return this.exportPolicyItem;
+        }
+
+        private ExportMenu build() {
+            this.exportNodeSourceItem = new MenuItem("Export Node Source", this.menu.exportItemImageResource);
+            Menu exportNodeSourceSubItems = new Menu();
+            MenuItem exportNodeSourceToFileItem = new MenuItem("To File", this.menu.exportItemImageResource);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "To File" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)